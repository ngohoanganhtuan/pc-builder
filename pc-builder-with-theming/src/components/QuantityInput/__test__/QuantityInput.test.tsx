import { render, fireEvent } from '@testing-library/react';
import * as renderer from 'react-test-renderer';

// Components
import { QuantityInput } from '@components';

describe('Test [QuantityInput] component', () => {
  test('Component [QuantityInput] render correctly', () => {
    const component = renderer
      .create(
        <QuantityInput
          value={500}
          onClickIncrease={jest.fn()}
          onClickDecrease={jest.fn()}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [QuantityInput] calls function when user click increase', () => {
    const handleIncrease = jest.fn();
    const { getByTestId } = render(
      <QuantityInput
        value={500}
        onClickIncrease={handleIncrease}
      />
    );
    fireEvent.click(getByTestId('increase-quantity-button'));
    expect(handleIncrease).toHaveBeenCalled();
  });

  test('Component [QuantityInput] calls function when user click decrease', () => {
    const handleDecrease = jest.fn();
    const { getByTestId } = render(
      <QuantityInput
        value={500}
        onClickDecrease={handleDecrease}
      />
    );
    fireEvent.click(getByTestId('decrease-quantity-button'));
    expect(handleDecrease).toHaveBeenCalled();
  });
});
