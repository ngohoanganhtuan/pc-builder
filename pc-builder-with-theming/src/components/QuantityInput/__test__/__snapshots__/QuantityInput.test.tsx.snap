// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [QuantityInput] component Component [QuantityInput] render correctly 1`] = `
<div
  className="quantity-input d-flex input-group-md shadow rounded-pill overflow-hidden quantity-input-md"
>
  <button
    className="btn bg-white shadow-none"
    data-testid="decrease-quantity-button"
    onClick={[MockFunction]}
  >
    <i
      className="bi bi-dash"
    />
  </button>
  <div
    className="number bg-white d-flex align-items-center w-100 justify-content-center"
  >
    500
  </div>
  <button
    className="btn bg-white shadow-none"
    data-testid="increase-quantity-button"
    onClick={[MockFunction]}
  >
    <i
      className="bi bi-plus"
    />
  </button>
</div>
`;
